test/verify/test_literal_limits.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2023 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "verify_program.hpp"
+#include <migraphx/program.hpp>
+#include <migraphx/make_op.hpp>
+#include <limits>
+
+struct test_literal_limits : verify_program<test_literal_limits>
+{
+    migraphx::program create_program() const
+    {
+        migraphx::program p;
+        auto* mm   = p.get_main_module();
+        auto input_s = migraphx::shape(migraphx::shape::float_type, {3, 1});
+        auto infinity_val = std::numeric_limits<float>::infinity();
+        std::vector<float> s_data{infinity_val, -infinity_val, std::numeric_limits<float>::quiet_NaN()};
+        auto input = mm->add_literal(migraphx::literal{input_s, s_data});
+        mm->add_instruction(migraphx::make_op("isnan"), input);
```

Comment:
This will all get const folded. Maybe we can multiply the numbers by the input parameter before passing it to `isnan`.

Reply:
Okay i'll try that.

Reply:
looks like that did the trick.

```
Run instruction: main:@0 = check_context::migraphx::gpu::context  -> float_type, {}, {}
Time: 0.00601ms, 0.00652ms
Run instruction: main:@1 = hip::hip_copy_literal[id=main:@literal:0] -> float_type, {3, 1}, {1, 1}
Time: 0.00632ms, 0.00679ms
Run instruction: output = @param:output -> bool_type, {3, 1}, {1, 1}
Time: 0.0009ms, 0.00134ms
Run instruction: test_input = @param:test_input -> float_type, {3, 1}, {1, 1}
Time: 0.00023ms, 0.00067ms
Run instruction: main:@4 = gpu::code_object[code_object=13632,symbol_name=mul_isnan_kernel,global=3,local=1024,](main:@1,test_input,output) -> bool_type, {3, 1}, {1, 1}
Time: 0.0245ms, 0.068711ms
Run instruction: main:@0 = check_context::migraphx::gpu::context  -> float_type, {}, {}
Time: 0.00053ms, 0.00104ms
Run instruction: main:@1 = hip::hip_copy_literal[id=main:@literal:0] -> float_type, {3, 1}, {1, 1}
Time: 0.00078ms, 0.00135ms
Run instruction: output = @param:output -> bool_type, {3, 1}, {1, 1}
Time: 0.00041ms, 0.00087ms
Run instruction: test_input = @param:test_input -> float_type, {3, 1}, {1, 1}
Time: 0.00026ms, 0.00079ms
Run instruction: main:@4 = gpu::code_object[code_object=13632,symbol_name=mul_isnan_kernel,global=3,local=1024,](main:@1,test_input,output) -> bool_type, {3, 1}, {1, 1}
Time: 0.00703ms, 0.048141ms

```

Reply:
Literal looks fine too here:

`Run instruction: main:@0 = @literal{inf, -inf, nan} -> float_type, {3, 1}, {1, 1}
Time: 0.00717ms, 0.00731ms`

Reply:
Updated and pushed